src/main/java/com/google/devtools/build/lib/bazel/bzlmod/Version.java

```diff
@@ -91,22 +91,26 @@ public record Version(
    * compared differently based on whether it's digits-only or not.
    */
   @AutoCodec
-  record Identifier(boolean isDigitsOnly, int asNumber, String asString)
+  record Identifier(boolean isDigitsOnly, long asNumber, String asString)
       implements Comparable<Identifier> {
     static Identifier from(String string) throws ParseException {
       if (Strings.isNullOrEmpty(string)) {
         throw new ParseException("identifier is empty");
       }
       if (string.chars().allMatch(Character::isDigit)) {
-        return new Identifier(true, Integer.parseInt(string), string);
+        try {
+          return new Identifier(true, Long.parseUnsignedLong(string), string);
+        } catch (NumberFormatException e) {
```

Comment:
Got this check:

This catch block catches an exception and re-throws another, but swallows the caught exception rather than setting it as a cause. This can make debugging harder.

@fmeum Can you please fix this?